### PR TITLE
Implement time out using multiprocessing calls

### PR DESCRIPTION
**Pull request by Saransh-cpp**
## Changes/Additions

1.  `random_plot_generator` is now being called using `multiprocessing.Process()` everywhere to look out for frozen simulations and restart them.
2. Added necessary code for `coverage.py` to detect and cover tests running in a sub-process (changes in workflows, a new file `sitecustomise.py` and a `.coveragerc` file).

Coverage now generates a different report for every subprocess and then these reports are joined using `coverage combine`.

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
image

This happened while running tests on another branch and though everything was working fine, the simulation was restarted because of the large number of cycles. What should be the timeout that I should keep for generating summary variables? Because I think this is the only simulation where 10 minutes would be less in some cases?

**Comment by rtimms** [comment]
Can you change the timeout based on what simulation you are going to run? The simplest things would be to try and pick some good times based on trial an error, and maybe we can do something more clever later.

File: bot/plotting/random_plot_generator.py [hunk 1]
```diff
@@ -182,35 +187,35 @@ def random_plot_generator(
...
-                        parameter_values,
-                        None,
-                        chemistry,
-                        solver,
-                        True,
-                        cycle_received,
-                        number,
-                        False,
-                    )
+                    return_dict["model"] = model
```
Comment (brosaplanella) [inline bot/plotting/random_plot_generator.py:190]: It is totally OK this way, but it might look cleaner if you use `return_dict.update()` instead.

Comment (brosaplanella) [reply bot/plotting/random_plot_generator.py:190]: (same applies to the other cases)

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks Saransh!

File: bot/plotting/random_plot_generator.py (hunk 1 above)
Comment (Saransh-cpp) [reply bot/plotting/random_plot_generator.py:190]: Updated in #10, thanks!

**Pull request merged**